tests/rados/test_osd_db_wal_configuration.py

```diff
@@ -0,0 +1,207 @@
+                osd_nodes.append(node)
+        test_count = 0
+
+        for osd_node in osd_nodes:
+            ssd_paths = available_ssd_path[osd_node.hostname]
+            for path in ssd_paths:
+                # Creating Volume group
+                cmd_vgcreate = f"vgcreate LVMTest {path}"
+                log.info(
+                    f"Creating volume group in {osd_node.hostname} osd node at {path}"
+                )
+                vg_out, _ = osd_node.exec_command(
+                    sudo=True, cmd=cmd_vgcreate, check_ec=False
+                )
+                log.info(f"The volume creation output is {vg_out}")
+                if "successfully created" not in vg_out:
+                    log.info(
+                        f"Volume group is not created in the {osd_node.hostname}.Skipping the further tests."
+                    )
+                    continue
+                log.info(
+                    f"Volume group is created at {osd_node.hostname} osd node at {path}"
+                )
+                # Create logical volume
+                log.info("Creating logical volumes")
+                cmd_lvcreate_base = "lvcreate -l 5 -n"
+                if "nvme" in path:
+                    cmd_lvcreate = f"{cmd_lvcreate_base} new_wal LVMTest -y"
+                else:
+                    cmd_lvcreate = f"{cmd_lvcreate_base} new_db LVMTest -y"
+                lv_out, _ = osd_node.exec_command(
+                    sudo=True, cmd=cmd_lvcreate, check_ec=False
+                )
+                log.info(f"Logical volumes are created in the {osd_node.hostname}")
+                if "created" not in lv_out:
+                    log.info(
+                        f"Logical volume  is not created in the {osd_node.hostname}.Skipping the further tests."
+                    )
+                    continue
+                # If in all nodes VG or lv creation fails then test_count is 0 and making the tes case fail.
+                test_count += 1
+                osds_id_in_node = rados_object.collect_osd_daemon_ids(osd_node)
+                test_osd = random.choice(osds_id_in_node)
+                log.info(f"Performing the configurations on {test_osd}")
+                # Get the OSD_ID
+                osd_fsid = rados_object.get_osd_fsid(test_osd)
+                log.info(f" The osd-{test_osd} fsid is - {osd_fsid}")
+                # stop test osd-
+                if not rados_object.change_osd_state(action="stop", target=test_osd):
+                    log.error(f"Unable to stop the OSD : {test_osd}")
+                    raise Exception("Execution error")
+                # Get the LVM list before the configuration
+                before_configure_lvm = rados_object.get_ceph_volume_lvm_list(
+                    osd_node, test_osd
+                )
+                log.info(
+                    f" ceph volume lvm list for the osd{test_osd} before configuration is {before_configure_lvm}"
```

Comment:
There's a bug, where the new Wal/db device added was not displayed in ceph-volume lvm list. This was via CBT.

Please check if it's the same case with ceph-volume as well.

Bug : https://bugzilla.redhat.com/show_bug.cgi?id=2237399

Reply:
Getting wal devices in the ceph-volume lvm list-

_[ceph: root@mero006 /]# ceph osd metadata osd.41
{
    "id": 41,
    "arch": "x86_64",
    "back_addr": "[v2:10.8.129.226:6876/331904443,v1:10.8.129.226:6877/331904443]",
    "back_iface": "",
    "bluefs": "1",
    "bluefs_dedicated_db": "0",
    **"bluefs_dedicated_wal":** "1",
    "bluefs_single_shared_device": "0",
    "bluefs_wal_access_mode": "blk",
    "bluefs_wal_block_size": "4096",
    "bluefs_wal_dev_node": "/dev/dm-12",
    "bluefs_wal_devices": "nvme0n1",
    "bluefs_wal_driver": "KernelDevice",
    "bluefs_wal_optimal_io_size": "0",
    "bluefs_wal_partition_path": "/dev/dm-12",
    "bluefs_wal_rotational": "0",
    "bluefs_wal_size": "1249902592",
    "bluefs_wal_support_discard": "1",
    "bluefs_wal_type": "ssd",
    "bluestore_bdev_access_mode": "blk",
    "bluestore_bdev_block_size": "4096",
    "bluestore_bdev_dev_node": "/dev/dm-10",
    "bluestore_bdev_devices": "sdk",
    "bluestore_bdev_driver": "KernelDevice",
    "bluestore_bdev_optimal_io_size": "262144",
    "bluestore_bdev_partition_path": "/dev/dm-10",
    "bluestore_bdev_rotational": "1",
    "bluestore_bdev_size": "8001016561664",
    "bluestore_bdev_support_discard": "0",
    "bluestore_bdev_type": "hdd",
    "bluestore_min_alloc_size": "4096",
    "ceph_release": "reef",
    "ceph_version": "ceph version 18.2.0-122.el9cp (cbcd542ab4c99bbd64f6264a3fb86e80ab335707) reef (stable)",
    "ceph_version_short": "18.2.0-122.el9cp",
    "ceph_version_when_created": "ceph version 18.2.0-122.el9cp (cbcd542ab4c99bbd64f6264a3fb86e80ab335707) reef (stable)",
    "container_hostname": "mero006",
    "container_image": "registry-proxy.engineering.redhat.com/rh-osbs/rhceph@sha256:8d9ae76a9369fb32a04506fd242b54e4cc3b3bc3f99d8cbc2186675e17de0cff",
    "cpu": "Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz",
    "created_at": "2023-11-15T05:41:01.981357Z",
    "default_device_class": "hdd",
    "device_ids": "nvme0n1=MTFDHBG960MCH-1AN1ZABYY_ZF1000XW,sdk=AVAGO_SMC3108_00194ccb191bd2fc2200436f23800403",
    "device_paths": "nvme0n1=/dev/disk/by-path/pci-0000:82:00.0-nvme-1,sdk=/dev/disk/by-path/pci-0000:04:00.0-scsi-0:2:8:0",
    "devices": "nvme0n1,sdk",
    "distro": "rhel",
    "distro_description": "Red Hat Enterprise Linux 9.3 (Plow)",
    "distro_version": "9.3",
    "front_addr": "[v2:10.8.129.226:6874/331904443,v1:10.8.129.226:6875/331904443]",
    "front_iface": "",
    "hb_back_addr": "[v2:10.8.129.226:6880/331904443,v1:10.8.129.226:6881/331904443]",
    "hb_front_addr": "[v2:10.8.129.226:6878/331904443,v1:10.8.129.226:6879/331904443]",
    "hostname": "mero006",
    "journal_rotational": "0",
    "kernel_description": "#1 SMP PREEMPT_DYNAMIC Fri Aug 25 09:13:12 EDT 2023",
    "kernel_version": "5.14.0-284.30.1.el9_2.x86_64",
    "mem_swap_kb": "0",
    "mem_total_kb": "131351044",
    "network_numa_unknown_ifaces": "back_iface,front_iface",
    "objectstore_numa_nodes": "1",
    "objectstore_numa_unknown_devices": "sdk",
    "os": "Linux",
    "osd_data": "/var/lib/ceph/osd/ceph-41",
    "osd_objectstore": "bluestore",
    "osdspec_affinity": "osd_spec_hdd",
    "rotational": "1"
}
[ceph: root@mero006 /]#_

```
====== osd.41 ======

  [wal]         /dev/LVMTest/new_wal

      block device              /dev/ceph-9cdad48f-fbd2-4871-8810-de5ed0bb623d/osd-block-1ddd8d2e-1c70-4b07-9261-34164f12a192
      block uuid                GZcH8Z-x2Gd-McjP-QEY1-1yWw-UHHX-4Aj1Ox
      cephx lockbox secret      
      cluster fsid              61c67698-8378-11ee-9bbc-ac1f6b40d3fc
      cluster name              ceph
      crush device class        
      encrypted                 0
      osd fsid                  1ddd8d2e-1c70-4b07-9261-34164f12a192
      osd id                    41
      osdspec affinity          osd_spec_hdd
      type                      wal
      vdo                       0
      wal device                /dev/LVMTest/new_wal
      wal uuid                  OylNBi-MCPH-hX0H-n7Ac-Xblw-rWLN-SdKtCc
      devices                   /dev/nvme0n1

  [block]       /dev/ceph-9cdad48f-fbd2-4871-8810-de5ed0bb623d/osd-block-1ddd8d2e-1c70-4b07-9261-34164f12a192

      block device              /dev/ceph-9cdad48f-fbd2-4871-8810-de5ed0bb623d/osd-block-1ddd8d2e-1c70-4b07-9261-34164f12a192
      block uuid                GZcH8Z-x2Gd-McjP-QEY1-1yWw-UHHX-4Aj1Ox
      cephx lockbox secret      
      cluster fsid              61c67698-8378-11ee-9bbc-ac1f6b40d3fc
      cluster name              ceph
      crush device class        
      encrypted                 0
      osd fsid                  1ddd8d2e-1c70-4b07-9261-34164f12a192
      osd id                    41
      osdspec affinity          osd_spec_hdd
      type                      block
      vdo                       0
      wal device                /dev/LVMTest/new_wal
      wal uuid                  OylNBi-MCPH-hX0H-n7Ac-Xblw-rWLN-SdKtCc
      devices                   /dev/sdk
```